data_prep/parse_training_data.py

```diff
@@ -198,10 +230,49 @@ def save_json(self, coverage: bool, group: bool, save_dir: str) -> None:
     data_filapath = os.path.join(save_dir, data_filename)
     with open(data_filapath, 'w') as file:
       json.dump(data, file, indent=4)
-    print(f'Saved to {data_filapath}.')
+    logging.info('Saved to: %s', data_filapath)
+
+
+def _download_files(experiment_dir: str, bucket_url: str) -> None:
+  """
+  Downloads files in |bucket_url| to |experiment_dir| and preserve their paths.
+  """
+  bucket_name = bucket_url.removeprefix('gs://').split('/')[0]
+  directory_prefix = bucket_url.removeprefix(f'gs://{bucket_name}/')
+  bucket = STORAGE_CLIENT.bucket(bucket_name)
+  blobs = bucket.list_blobs(prefix=directory_prefix)
```

Comment:
how important is the count? this *may* be a little inefficient in that it requires iterating/listing the bucket twice. not sure if there's any caching happening under the hood.

Reply:
You are right: it is inefficient to list the bucket twice.
This count is only a personal preference to help me visualize the progress while downloading them (e.g., `i / count`) because a full experiment has many files to download. If there are too many files or it is taking too long, I can kill this and consider a different approach.

Happy to remove it if you think that's not necessary, or I can edit it if there is a better approach.
Thanks!